jc/parsers/route_print.py

```diff
@@ -0,0 +1,565 @@
+            "network_destination": "::1/128",
+            "gateway": "On-link"
+          },
+          {
+            "interface": 12,
+            "metric": "271",
+            "network_destination": "2001:db8::/64",
+            "gateway": "fe80::1"
+          },
+          {
+            "interface": 12,
+            "metric": "271",
+            "network_destination": "fdb2:2c26:f4e4::/64",
+            "gateway": "On-link"
+          },
+          {
+            "interface": 12,
+            "metric": "271",
+            "network_destination": "fdb2:2c26:f4e4:0:670f:fec7:75b7:eb83/128",
+            "gateway": "On-link"
+          },
+          {
+            "interface": 12,
+            "metric": "271",
+            "network_destination": "fdb2:2c26:f4e4:0:b4d5:8083:6ce6:37d/128",
+            "gateway": "On-link"
+          },
+          {
+            "interface": 12,
+            "metric": "271",
+            "network_destination": "fe80::/64",
+            "gateway": "On-link"
+          },
+          {
+            "interface": 12,
+            "metric": "271",
+            "network_destination": "fe80::4abd:fa70:3d36:83fd/128",
+            "gateway": "On-link"
+          },
+          {
+            "interface": 1,
+            "metric": "331",
+            "network_destination": "ff00::/8",
+            "gateway": "On-link"
+          },
+          {
+            "interface": 12,
+            "metric": "271",
+            "network_destination": "ff00::/8",
+            "gateway": "On-link"
+          }
+        ],
+        "persistent_routes": [
+          {
+            "interface": 0,
+            "metric": "4294967295",
+            "network_destination": "2001:db8::/64",
+            "gateway": "fe80::1"
+          }
+        ]
+      }
+    }
+"""
+
+import re
+import jc.utils
+
+
+class info():
+    """Provides parser metadata (version, author, etc.)"""
+    version = '1.0'
+    description = '`route-print` command parser'
+    author = 'joehacksalot'
+    author_email = 'joehacksalot@gmail.com'
+    compatible = ['windows']
+    magic_commands = ['route-print']
+    tags = ['command']
+
+
+__version__ = info.version
+
+
+def parse(data, raw=False, quiet=False):
+    """
+    Main text parsing function
+
+    Parameters:
+
+        data:        (string)  text data to parse
+        raw:         (boolean) unprocessed output if True
+        quiet:       (boolean) suppress warning messages if True
+
+    Returns:
+
+        Parsed dictionary. The raw and processed data structures are the same.
+    """
+    jc.utils.compatibility(__name__, info.compatible, quiet)
+    jc.utils.input_type_check(data)
+
+    raw_output = {}
+    if jc.utils.has_data(data):
+        try:
+            raw_output = _parse(data)
+            return raw_output if raw else _process(raw_output)
+        except Exception as e:
+            if not quiet:
+                jc.utils.warning_message(['Could not parse data due to unexpected format.'])
```

Comment:
This type of exception handling can make it difficult to troubleshoot. Doing this means the user cannot use `jc --route-print -dd` to get debug info. Without the exception handling `jc` should just do the right thing. If you want a custom exception message, it would be better to use `jc.exceptions.ParseError`:

```
from jc.exceptions import ParseError
raise ParseError('My custom error here.')
```
You can find examples in the following parsers: cef.py, csv.py, uname.py, etc.